pypesto/ensemble/ensemble.py

```diff
@@ -1253,3 +1270,77 @@ def calculate_cutoff(
 
     range = chi2.ppf(q=percentile / 100, df=df)
     return fval_opt + range
+
+
+def calculate_hpd(
+    result: Result,
+    burn_in: int = 0,
+    ci_level: float = 0.95,
+):
+    """
+    Calculate Highest Posterior Density (HPD) samples of pypesto sampling result.
+
+    The HPD is calculated for a user-defined credibility level (ci_level). The
+    HPD includes all parameter vectors with a (non-normalized) posterior
+    probability that is higher than the lowest 1-ci_level %
+    posterior probability values.
+
+    Parameters
+    ----------
+    result:
+        The optimization result from which to create the ensemble.
+    burn_in:
+        Burn in index that is cut off before HPD is calculated.
+    ci_level:
+        Credibility level of the resulting HPD. 0.95 corresponds to the 95% CI.
+        Values between 0 and 1 are allowed.
+
+    Returns
+    -------
+    The HPD parameter vector.
+    """
+    if ci_level < 0 or ci_level > 1:
```

Comment:
Matches the error message now.
```suggestion
    if not 0 <= ci_level <= 1:
```